helpers/ed.js

```diff
@@ -28,22 +28,23 @@ ed.makeKeypair = function (hash) {
  * Creates a signature based on a hash and a keypair.
  * @implements {sodium}
  * @param {hash} hash
- * @param {keypair} keypair
- * @return {signature} signature
+ * @param {Buffer} privateKey
+ * @return {Buffer} signature
  */
-ed.sign = function (hash, keypair) {
-	return sodium.crypto_sign_detached(hash, Buffer.from(keypair.privateKey, 'hex'));
+ed.sign = function (hash, privateKey) {
```

Comment:
You are right, I can revert those changes. I added corresponding unit tests to the module